libs/auth/src/cryptography.ts

```diff
@@ -474,23 +474,34 @@ export function parseSignMessageInputExcludingMessage(
 export async function signMessageHelper({
   signingPrivateKey,
 }: SignMessageInputExcludingMessage): Promise<Buffer> {
-  const signParams: OpensslParam[] = (() => {
+  // While it's possible to hash and sign with a single OpenSSL command, we separate the two
+  // actions. Hashing large inputs with the TPM can be extremely slow. So in production, we hash
+  // using the main processor and only sign using the TPM.
+  const hash = await openssl(['dgst', '-sha256', '-binary'], {
+    stdin: process.stdin,
+  });
```

Comment:
Is this because we're trying to keep it as similar as possible to what it was before? I don't see anything here that indicates why it isn't just using `node:crypto`, so I'd be tempted to make that change in the future. This doesn't provide any context on that.

Reply:
I would suggest keeping the hash in OpenSSL so all of this can be done in the FIPS-compliant component.

Reply:
Add a comment to that effect?

Reply:
Comment updated: https://github.com/votingworks/vxsuite/pull/4816/commits/4b9fb5ed69af331f0b94471110870b43278720f1